### PR TITLE
CI: Moving CircleCI build to Travis

### LINKED ISSUE #23821: CI: Move CircleCI job to Azure pipelines
We've been moving our CI from other systems to Travis and Azure Pipelines, which makes things simpler, as we don't need to maintain several CI configuration files, and check for CI errors and logs in different environments.

We have almost everything in those two systems now, except one last job that is in CircleCI. This job requires PostgreSQL and MySQL databases, which hasn't been configured yet in Azure Pipelines. So, some research on how to set up them will be required.

In this issue we should create a new job in Azure Pipelines `ci/azure/linux.yml` equivalent to the one in CircleCI, and remove all CircleCI files `.circleci`.

CC: @TomAugspurger

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] xref #23821
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Moving the CircleCI job to Travis. In a separate PR, I'll some jobs not using mysql and postgres to Azure, to balance things, and make the whole CI finish earlier.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,16 +43,23 @@ matrix:
...
           - language-pack-zh-hans
+
     - dist: trusty
       env:
         - JOB="2.7" ENV_FILE="ci/deps/travis-27.yaml" PATTERN="not slow"
       addons:
         apt:
           packages:
           - python-gtk2
+
```
Comment (jreback) [inline .travis.yml]: can you also add this in the allowed failures section, i am a little leary of adding a 6th job

Comment (datapythonista) [reply .travis.yml]: I'm planning to move the jobs (or some of them) not using mysql or postgres to azure after this PR.

I can surely move it to allowed failures, but we shouldn't have 6 jobs for more than couple of days, if that makes a difference.

Comment (jreback) [reply .travis.yml]: ok that’s fine the

Comment (datapythonista) [reply .travis.yml]: The CI is green here, I expected circle to not run, as it's been removed from the repo. But it just looks as queued. I guess it won't appear when the config is deleted from master, or we may have to unlink our disable our account or something.

**Comment by jreback** [comment]
thanks. will disable Circle now.

**Pull request merged**